src/element-service.js

```diff
@@ -207,25 +207,26 @@ export function extensionScriptsInNode(head) {
  * Waits for an extension if its script is present
  * @param {!Window} win
  * @param {string} extension
- * @param {!Array<string>} customElements
+ * @param {HTMLHeadElement|Element|ShadowRoot} head
  * @return {!Promise}
  * @private
  */
-function waitForExtensionIfPresent(win, extension, customElements) {
+function waitForExtensionIfPresent(win, extension, head) {
   /**
    * If there is an extension script wait for it to load before trying
    * to get the service. Prevents a race condition when everything but
    * the extensions is in cache. If there is no script then it's either
    * not present, or the service was defined by a test. In those cases
    * we don't wait around for an extension that does not exist.
    */
-
-  if (customElements.includes(extension)) {
-    const extensions = getService(win, 'extensions');
-    return /** @type {!Promise<?Object>} */ (
-      extensions.waitForExtension(win, extension));
+  if (isElementScheduled(win, extension) ||
```

Comment:
`!isElementScheduled`. If the element is scheduled, we know we have to `waitForExtension`.